sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -178,9 +195,24 @@ public TableClient(string connectionString, string tableName, TableClientOptions
 
         internal TableClient(Uri endpoint, string tableName, TableSharedKeyPipelinePolicy policy, AzureSasCredential sasCredential, TableClientOptions options)
         {
-            Argument.AssertNotNull(tableName, nameof(tableName));
             Argument.AssertNotNull(endpoint, nameof(endpoint));
 
+            // If we were provided no tableName, try to parse it from the endpoint.
```

Comment:
What if the endpoint and credential both specify a table name and they don't agree?

Reply:
It will result in a 403